CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+# Contributing
+
+First, thank you for contributing to Iceberg Rust! The goal of this document is to provide everything you need to start contributing to iceberg-rust. The following TOC is sorted progressively, starting with the basics and expanding into more specifics.
+
+- [Your First Contribution](#your-first-contribution)
+- [Workflow](#workflow)
+  - [Git Branches](#git-branches)
+  - [GitHub Pull Requests](#github-pull-requests)
+    - [Title](#title)
+    - [Reviews & Approvals](#reviews--approvals)
+    - [Merge Style](#merge-style)
+    - [CI](#ci)
+- [Setup](#setup)
+  - [Using a dev container environment](#using-a-dev-container-environment)
+  - [Bring your own toolbox](#bring-your-own-toolbox)
+- [Code of Conduct](#code-of-conduct)
+
+## Your First Contribution
+
+1. Ensure your change has an issue! Find an [existing issue](https://github.com/apache/iceberg-rust/issues) or [open a new issue](https://github.com/apache/iceberg-rust/issues/new).
```

Comment:
For me, it feels redundant to have a mandatory issue for each of the PRs. Do we want to make this a hard requirement?

Reply:
Removed.